sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java

```diff
@@ -209,4 +286,52 @@ protected void sleepIfRunningAgainstService(long millis) {
             throw logger.logExceptionAsWarning(new IllegalStateException(ex));
         }
     }
+
+    /**
+     * Get offset which determines the rolling strategy.
+     *
+     * @return The offset according to day of the week and platform information
+     */
+    private static int getOffset() {
+        if (testMode == TestMode.PLAYBACK) {
+            return 0;
+        }
+        LocalDate today = LocalDate.now(ZoneId.systemDefault());
+        return (today.getDayOfWeek().getValue() + getPlatFormOffset()) % PLATFORM_LIST.size();
+    }
+
+    private static Integer getPlatFormOffset() {
+        ClientLogger logger = new ClientLogger(TestBase.class);
+        String currentOs = System.getProperty("os.name").toLowerCase(Locale.ROOT);
+        logger.info("It is currently running on os: " + currentOs);
+        String currentJdk = System.getProperty("java.version").toLowerCase(Locale.ROOT);
+        logger.info("It is using jdk: " + currentJdk);
+
+        for (int i = 0; i < PLATFORM_LIST.size(); i++) {
+            if (isCurrentPlatformIncluded(currentOs, currentJdk, i)) {
+                return i;
+            }
+        }
+        throw new RuntimeException(String.format("Not running on the expected platform. os: %s, jdk: %s",
+            currentOs, currentJdk));
+    }
+
+    private static boolean isCurrentPlatformIncluded(String currentOs, String currentJdk, int pos) {
+        boolean flag = currentOs.toLowerCase(Locale.ROOT).startsWith(PLATFORM_LIST.get(pos).split(",")[0].trim()
+            .toLowerCase(Locale.ROOT));
+        flag = flag && Arrays.stream(PLATFORM_LIST.get(pos).split(",")[1].trim().split("\\|")).anyMatch(
+            comparedJdk -> currentJdk.toLowerCase(Locale.ROOT).startsWith(comparedJdk.trim().toLowerCase(Locale.ROOT))
+        );
+        return  flag;
+    }
+
+    private static List<String> buildPlatformList() {
```

Comment:
This information should not be included in the test library. The platforms should be part of the pipeline yml file as that is how we should be configuring the platforms for our test runs.